pkg/machinehealthcheck/reconciler.go

```diff
@@ -0,0 +1,101 @@
+package reconciler
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	anywherecluster "github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/clusterapi"
+	"github.com/aws/eks-anywhere/pkg/constants"
+	"github.com/aws/eks-anywhere/pkg/controller/clientutil"
+	"github.com/aws/eks-anywhere/pkg/controller/serverside"
+)
+
+// Reconciler allows to reconcile machine health checks.
+type Reconciler struct {
+	client client.Client
+}
+
+// New returns a new Reconciler.
+func New(client client.Client) *Reconciler {
```

Comment:
Can we initialize the defaulter here through the factory? I think that's what we do for cli right? And then we can just use the same defaulter instead of calling the exported method below.